manifests/04-ca-config.yaml

```diff
@@ -1,8 +0,0 @@
----
-apiVersion: v1
-kind: ConfigMap
```

Comment:
you should check w/ @abhinavdahiya what's going to happen here when this resource vanishes from the manifest in an upgrade scenario.

seems like it would be simpler to just mark it as a create-only resource.

Reply:
it's not obvious to me what we buy by having the serviceca generator handle this resource.

Reply:
> you should check w/ @abhinavdahiya what's going to happen here when this resource vanishes from the manifest in an upgrade scenario.

Per Abhinav the CVO will leave the object as-is, so on upgrade the existing generator code will take over managing state.

> it's not obvious to me what we buy by having the serviceca generator handle this resource.

Maybe a minor benefit - if the operator state is set to `Removed`, we can delete the ConfigMap as the registry operator does not consume the service CA.

Reply:
> Per Abhinav the CVO will leave the object as-is, so on upgrade the existing generator code will take over managing state.

keep in mind you don't really want the generator doing it either.  you want the injector doing it.

now that said, the generator isn't touching the keys inside the configmap, so it's *probably* ok.

Reply:
We have precedent in ocm-o, where the operator creates and manages the service CA ConfigMap and object meta: https://github.com/openshift/cluster-openshift-controller-manager-operator/blob/701ad56928ab79a26b696bf3fe66510216e172dd/pkg/operator/sync_openshiftcontrollermanager_v311_00.go#L198-L228